### PR TITLE
Use Java 17 for build and runtime

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Use Java 17 for build and runtime. Enforce Java 17+ in build.gradle. Use Java 17 in CircleCI config. Remove Java 11 variant of docker.

## Fixed Issue(s)

#718 
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -55,15 +55,15 @@ executors:
...
+      image: ubuntu-2004:current # https://circleci.com/developer/machine/image/ubuntu-2004
     resource_class: arm.medium
     environment:
       architecture: "arm64"
       platform: "linux/arm64"
 
   machine_executor_amd64:
     machine:
-      image: ubuntu-2004:202201-02 # Ubuntu 20.04, Docker v20.10.12, Docker Compose v1.29.2
+      image: ubuntu-2004:current # https://circleci.com/developer/machine/image/ubuntu-2004
```
Comment (jframe) [inline .circleci/config.yml:66]: We are using quite an old image, ubuntu-2204 would be better to use

File: gradle/license-report-config/license-normalizer.json [hunk 2]
```diff
@@ -7,6 +7,7 @@
   ],
   "transformationRules" : [
     { "bundleName" : "MIT", "licenseNamePattern" : "\"MIT\\sLicense\"" },
+    { "bundleName" : "MIT", "licenseNamePattern" : "(.*)SPDX-License-Identifier: MIT(.*)" },
```
[the remark this answers is not in the document]
Comment (usmansaleem) [reply gradle/license-report-config/license-normalizer.json:10]: No, this was resulted due to some other library and was spitting warning for quite some time.

File: .circleci/config.yml (hunk 1 above)
Comment (usmansaleem) [reply .circleci/config.yml:66]: 👍🏼 sure. easy change to make.

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**